### PR TITLE
Auctions Page

### PR DESCRIPTION
**Pull request by swarajpure**
Fixes https://github.com/Real-Dev-Squad/website-todo-items/issues/16

Unable to add GIF preview here due to size limitation. Please check here: https://cdn.discordapp.com/attachments/748267291405320204/820029342876827658/auctions_gif.gif

### CODE REVIEW

File: components/Auctions/createNewAuction.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      <form className={styles.newAuctionForm} onSubmit={auctionSubmitHandler}>
+        {isSubmitted && <p>Please wait...</p>}
+        <label name="itemType">
+          Item to auction:
+          <br />
+          <input
+            className={styles.inputBox}
+            htmlFor="itemType"
+            onChange={(e) => setItemType(e.target.value)}
+            required
```
Comment (sumitd94) [inline components/Auctions/createNewAuction.js:63]: the `htmlFor` should be used in label right? and there you need to pass the reference of this input(using id)

File: components/Auctions/createNewAuction.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+          />
+        </label>
+        <label name="quantity">
+          Quantity for auction:
+          <br />
+          <input
+            className={styles.inputBox}
+            htmlFor="quantity"
+            onChange={(e) => setQuantity(e.target.value)}
```
Comment (sumitd94) [inline components/Auctions/createNewAuction.js:72]: NIT: you can destruture the `e` here
```javascript
onChange={({ target: { value } }) => setQuantity(value)}
```

File: components/Auctions/createNewAuction.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            required
+          />
+        </label>
+        <label name="initialPrice">
+          Initial price for this auction:
+          <br />
+          <input
+            className={styles.inputBox}
+            htmlFor="initialPrice"
+            onChange={(e) => setInitialPrice(e.target.value)}
```
Comment (sumitd94) [inline components/Auctions/createNewAuction.js:82]: NIT: you can destruture the `e` here
```javascript
onChange={({ target: { value } }) => setInitialPrice(value)}
```

File: components/Auctions/createNewAuction.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            required
+          />
+        </label>
+        <label name="duration">
+          Duration of auction (in number of days):
+          <br />
+          <input
+            className={styles.inputBox}
+            htmlFor="duration"
+            onChange={(e) => setDuration(e.target.value)}
```
Comment (sumitd94) [inline components/Auctions/createNewAuction.js:92]: NIT: you can destruture the `e` here
```javascript
onChange={({ target: { value } }) => setDuration(value)}
```

File: components/Auctions/index.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+          <form
+            className={styles.bidOptions}
+            onSubmit={(e, id) => handleNewBid(e, id)}
+          >
+            <input
+              className={styles.inputBid}
+              type="number"
+              min={parseInt(highest_bid) + 1}
+              required="true"
+              onBlur={(e) => validateBid(e.target.value, highest_bid)}
```
Comment (sumitd94) [inline components/Auctions/index.js:101]: NIT: Destructure the `e`

File: components/Auctions/index.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            <h2>
+              Current Bid:
+              <div className={styles.currentBidPrice}>{highest_bid}</div>
+            </h2>
+          </div>
+        </div>
+        <div>
+          <form
+            className={styles.bidOptions}
+            onSubmit={(e, id) => handleNewBid(e, id)}
```
Comment (sumitd94) [inline components/Auctions/index.js:94]: NIT: Destructure the `e`

**Review by sumitd94 — CHANGES_REQUESTED** [review]

**Comment by SBagaria2710** [comment]
LGTM 👍🏼

**Review by swarajpure — COMMENTED** [review]
@sumitd94 Done with the changes. Please check

File: components/Auctions/Auctions.module.css [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+@media only screen and (max-width: 500px) {
+  .auctionContainer {
+    width: 90%;
+  }
+/* 
+  .bidBtn {
+    width: 100px;
+  } */
```
Comment (sumitd94) [inline components/Auctions/Auctions.module.css:130]: NIT: Can we please remove this if not needed.

**Review by sumitd94 — APPROVED** [review]
Great job 🔥

File: components/Auctions/index.js [hunk 8]
```diff
@@ -0,0 +1,156 @@
+import Image from 'next/image';
+import React, { useState, useEffect } from 'react';
+import styles from './Auctions.module.css';
+import fetchData from '../../utils/fetchData';
+import fetchSelfDetails from '../../utils/fetchSelfDetails';
+
+const BASE_API_URL = process.env.NEXT_PUBLIC_BASE_API_URL;
+const AUCTIONS_URL = `${BASE_API_URL}/auctions`;
+const BASE_IMAGE_URL =
```
Comment (harshith-venkatesh) [inline components/Auctions/index.js:9]: Would it be better to have it in .env so that it can be used everywhere.... Just a thought

Comment (ankushdharkar) [reply components/Auctions/index.js:9]: @harshith-venkatesh Why not a file constant?

File: components/Auctions/Auctions.module.css [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+}
+.biddersImg[data-columns="5"]{
+  width: calc(160px/5);
+}
+.biddersImg[data-columns="6"]{
+  width: calc(160px/6);
+}
+.biddersImg[data-columns="7"]{
+  width: calc(160px/7);
+}
```
Comment (ankushdharkar) [inline components/Auctions/Auctions.module.css:109]: What is this doing?

File: components/Auctions/index.js [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+  const fetchAndSetAuctions = async () => {
+    const response = await fetchData(`${AUCTIONS_URL}`);
+    const json = await response.json();
+    setAuctionsData(json.auctions);
+    setIsLoading(false);
+  };
+
+  const handleNewBid = async (e, auctionId) => {
+    if (!isUserLoggedIn) {
+      return alert('Please log in to bid!');
```
Comment (ankushdharkar) [inline components/Auctions/index.js:43]: NIT:
```
alert();
return;
```

File: components/Auctions/index.js [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+    }
+  };
+
+  const validateBid = (userBid, highestBid) => {
+    if (userBid <= highestBid) alert('You must bid higher than current bid!');
+    else setUserBid(userBid);
+  };
+
+  const getColumns = (totalBidder) => {
+    return totalBidder <= 12 ? Math.ceil(totalBidder / 2) : 7;
```
Comment (ankushdharkar) [inline components/Auctions/index.js:72]: What are 12 and 7?

File: components/Auctions/index.js (hunk 8 above)
Comment (harshith-venkatesh) [reply components/Auctions/index.js:9]: That would be even better
I have observed this in status site, that we have gone with .env 
That's why I was biased with .env

File: components/Auctions/Auctions.module.css (hunk 9 above)
Comment (ice-rahul) [reply components/Auctions/Auctions.module.css:109]: Here we are changing the size of the bidder's image
Size of the bidder's Image-based on the number of columns allowed in the container horizontally.

File: components/Auctions/index.js (hunk 11 above)
Comment (ice-rahul) [reply components/Auctions/index.js:72]: MAX_ALLOWED_COLUMNS=7
MAX_BIDDERS_ALLOWED_BELOW_MAX_COLUMNS=12

**Review by harshith-venkatesh — APPROVED** [review]

File: components/Auctions/Auctions.module.css (hunk 9 above)
Comment (ankushdharkar) [reply components/Auctions/Auctions.module.css:109]: Why not use a declarative class or an inline sizing?

**Comment by ankushdharkar** [comment]
If all comments are resolved, time to ship this! 🚢

**Pull request merged**